frontend/src/pages/hardwareProfiles/nodeResource/NodeResourceForm.tsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { TextInput, FormGroup, Form } from '@patternfly/react-core';
+import { Identifier } from '~/types';
+import { UpdateObjectAtPropAndValue } from '~/pages/projects/types';
+import { UnitOption } from '~/utilities/valueUnits';
+import { validateDefaultCount, validateMinCount } from './utils';
+import CountFormField from './CountFormField';
+
+type NodeResourceFormProps = {
+  identifier: Identifier;
+  setIdentifier: UpdateObjectAtPropAndValue<Identifier>;
+  unitOptions?: UnitOption[];
+  existingIdentifier?: boolean;
+  isUniqueIdentifier?: boolean;
+};
+
+const NodeResourceForm: React.FC<NodeResourceFormProps> = ({
+  identifier,
+  setIdentifier,
+  unitOptions,
+  existingIdentifier,
+  isUniqueIdentifier,
+}) => {
+  const [validated, setValidated] = React.useState<'default' | 'error' | 'success'>('default');
+
+  React.useEffect(() => {
+    setValidated(isUniqueIdentifier ? 'default' : 'error');
+  }, [isUniqueIdentifier]);
```

Comment:
You don't need these at all:
```suggestion
  const validated = isUniqueIdentifier ? 'default' : 'error';
```